File: cryptography/src/main/java/com/personal/image_toolkit/PrivateKeyEntryDataInterface.java
package com.personal.image_toolkit;

import java.security.PrivateKey;
import java.security.cert.Certificate;

/**
 * Defines the contract for accessing data related to a PrivateKeyEntry.
 */
public interface PrivateKeyEntryDataInterface {

    /**
     * @return The alias of the key entry.
     */
    String getAlias();

    /**
     * @return The private key.
     */
    PrivateKey getPrivateKey();

    /**
     * @return The certificate chain.
     */
    Certificate[] getCertificateChain();

    /**
     * @return The password for the key.
     */
    char[] getKeyPassword();

    /**
     * Clears the key password from the internal array for security best practice.
     */
    void clearPassword();
}

File: cryptography/src/main/kotlin/com/personal/image_toolkit/PrivateKeyEntryDataInterface.kt
package com.personal.image_toolkit

import java.security.PrivateKey
import java.security.cert.Certificate

/**
 * Defines the contract for accessing data related to a PrivateKeyEntry.
 */
interface PrivateKeyEntryDataInterface {
    /**
     * @return The alias of the key entry.
     */
    val alias: String

    /**
     * @return The private key.
     */
    val privateKey: PrivateKey

    /**
     * @return The certificate chain.
     */
    val certificateChain: Array<Certificate>

    /**
     * @return The password for the key.
     */
    val keyPassword: CharArray

    /**
     * Clears the key password from the internal array for security best practice.
     */
    fun clearPassword()
}